### PR TITLE
Added cli, better readme and support for starting and stopping sessions

### PR DESCRIPTION
**Pull request by adirajagopal**
- Added a cli to start and stop sessions
- Added API to start and stop sessions
- Made better examples/ dir  + README
- Better error handling and reporting on model_health checks and start/stop sessions.

### CODE REVIEW

File: ncompass/client/functional/client.py [hunk 1]
```diff
@@ -1,13 +1,43 @@
...
 from .run_prompt import stream_prompt
+from ncompass.network_utils import get
 from .model_health import check_model_health
+from ncompass.errors import model_not_started
+
+def start_stop_handler(response):
+    if (response.status_code == 200) or (response.status_code == 209): 
+        return True
+    elif (response.status_code == 400): 
+        raise RuntimeError('Internal server error, contact admin@ncompass.tech')
```
Comment (diederik-vink) [inline ncompass/client/functional/client.py:13]: wrong code. intenral server error is standarzied to  500.

File: ncompass/client/functional/client.py [hunk 2]
```diff
@@ -1,13 +1,43 @@
...
+    elif response.status_code == 504:
+        model_not_started(api_key)
+    else:
+        return False
 
 def wait_until_model_running(url, api_key):
-    while not model_is_running(url, api_key):
-        continue
+    break_loop = False
+    wait_until = datetime.now() + timedelta(seconds=10)
```
Comment (diederik-vink) [inline ncompass/client/functional/client.py:36]: 10 seconds feels kinda high no? Maybe more like 1-2?

File: ncompass/errors.py [hunk 3]
```diff
@@ -10,3 +10,9 @@ def api_key_not_set(custom_var = None):
         raise RuntimeError(\
                 'Env variable NCOMPASS_API_KEY needs to be set to the api key. ' 
                 'Alternatively, pass a custom variable.')
+
+def model_not_started(api_key):
+    msg = (f'Model {api_key} has likely not been started or is starting. '
```
Comment (diederik-vink) [inline ncompass/errors.py:15]: Why is our message unsure? I think we should say "Model {api_key} has not been started or is currently starting. '

File: ncompass/network_utils.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import httpx
+import aiohttp
+
+def exec_url():
+    return 'api.ncompass.tech'
+
+def get(url, headers=None):
+    client = httpx.Client(timeout=20, verify=False)
```
Comment (diederik-vink) [inline ncompass/network_utils.py:8]: i like the 20, just wondering where its from? have we tested this for 20 being reasonable? or just felt good?

**Review by diederik-vink — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -14,37 +14,42 @@ key, but for now please email either aditya.rajagopal@ncompass.tech or diederik.
...
 ```
 The section below unwraps the code a bit more. It is still simple, but not quite OLOC'd.
 
 ## More fine grained control 
-The following code (also found in examples/hello_world.py) sets up an nCompass client, parameters 
-and calls the complete_prompt API. Our API returns a streaming iterator (response_iterator) and 
-we also provide a print_prompt function which can print out the stream.
+The following code (also found in examples/fine_grained_control.py) sets up an nCompass client, 
+parameters and calls the complete_prompt API. Our API returns a streaming iterator 
+(response_iterator) and we also provide a print_prompt function which can print out the stream.
```
[the remark this answers is not in the document]
Comment (adirajagopal) [reply README.md:37]: Will change to this,  add a blocking parameter to client.

File: README.md [hunk 6]
```diff
@@ -53,3 +58,21 @@ key to the client directly as below.
 ```
 client = nCompass(api_key = '<key we provide you>')
 ```
+
+## **IMPORTANT:** What are sessions?
+Sessions are how nCompass keeps track of usage statistics. nCompass billing is **not per_token** 
+but rather **time_based**. This means, your costs to use this does not scale linearly with your
```
[the remark this answers is not in the document]
Comment (adirajagopal) [reply README.md:64]: this means your costs do not scale ...

File: examples/fine_grained_control.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
+from ncompass.client import nCompass
+
+def fine_grained_control():
+    # The client will automatically read the API key from the environment variable passed here.
+    client = nCompass(custom_env_var = 'NCOMPASS_API_KEY')
```
[the remark this answers is not in the document]
Comment (adirajagopal) [reply examples/fine_grained_control.py:5]: can also directly provides api key (see readme)...

File: ncompass/client/functional/client.py (hunk 2 above)
Comment (adirajagopal) [reply ncompass/client/functional/client.py:36]: I'll drop it 2.

File: ncompass/errors.py (hunk 3 above)
Comment (adirajagopal) [reply ncompass/errors.py:15]: yeah will change

File: README.md [hunk 8]
```diff
@@ -14,37 +14,42 @@ key, but for now please email either aditya.rajagopal@ncompass.tech or diederik.
...
 for i in range(3):
-    ttft = nCompassOLOC().complete_prompt('<api_key>', prompt, max_tokens = 300, temperature = 0.5
-                                          , top_p = 0.9, stream = True, pprint = True)
-    print(f'ttft = {ttft*1000:.2f}ms')
+    nCompassOLOC().start_session(api_key)
+    nCompassOLOC().complete_prompt(api_key, prompt, max_tokens = 300, temperature = 0.5
+                                   , top_p = 0.9, stream = True, pprint = True)
+    nCompassOLOC().stop_session(api_key)
 ```
 The section below unwraps the code a bit more. It is still simple, but not quite OLOC'd.
```
[the remark this answers is not in the document]
Comment (adirajagopal) [reply README.md:32]: remove'd ;) the apostrophe

**Pull request merged**